### DISCUSSION TITLE
Car routing avoiding toll roads per country

### DISCUSSION THREAD

**Discussion by eumiro** [organicmaps/organicmaps#4571] [Ideas]
Currently it is possible to avoid toll roads or not avoid them.

I would like to select a list of countries where I do not want to avoid toll roads, because I have already paid for their usage.

In a few European countries: Austria, Bulgaria, Czechia, Hungary, Moldova, Romania, Slovakia, Slovenia, and Switzerland ( see https://en.wikipedia.org/wiki/Vignette_(road_tax) ), one can buy a vignette for the motorways (usually valid for a year, but there may be also some shorter and cheaper intervals, such as 2 months or 10 days) and does not have to pay anything else per km/passage. However when one visits such a country, they might not want to buy such a vignette and thus want to avoid toll roads. When a motorway crosses a border, in some cases the toll road starts at the border, in other cases it is free until the first exit. Having the possibility to select this behavior per country would make it more useful for cross-border routing.

All it would need is to extend the menu “avoid toll roads” with a submenu containing these nine European countries. If there are any further countries with similar system, or if any of these countries abandons this system, someone will already post an issue/PR here. This submenu would have the first main checkbox “avoid all toll roads” and then text line “except in:” and a list of these nine checkboxes with country names. If the main checkbox is True, then the other nine checkboxes are selectable. If the main checkbox is False, then the other nine checkboxes are grayed out. The choice would be remembered on the device. 

Google Maps does not have this feature. Waze does have it. Viamichelin always counts the shortest vignette (e.g. 10 days) into the price of the trip and thus makes motorway trips more expensive than country road trips even for holders of yearly vignettes.

**Comment by biodranik** [comment]
Is it possible to buy vignettes online before the trip? It would be useful to provide this info for any travelers in advance. There was a related issue: https://github.com/organicmaps/organicmaps/issues/4462
